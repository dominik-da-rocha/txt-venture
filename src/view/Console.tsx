import "./Console.css";
import React from "react";
import Component, { ComponentProps, ComponentState } from "./Component";

export interface ConsoleProps extends ComponentProps {
  logBook: string[];
}

export interface ConsoleState extends ComponentState {}

export class Console extends Component<ConsoleProps, ConsoleState> {
  constructor(props: ConsoleProps) {
    super(props, {});
  }

  render() {
    return (
      <div id={this.props.id} className={this.className("Console")}>
        {this.props.logBook}
      </div>
    );
  }
}

export default Console;
