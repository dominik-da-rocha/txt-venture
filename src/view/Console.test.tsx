import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Renderer from 'react-test-renderer';
import React from "react";
import Console from "./Console";

Enzyme.configure({ adapter: new Adapter() });


describe("Console", () => {

  it("renders without crashing", () => {
    expect(() => {
      shallow(<Console id="test" logBook={[]} />);
    }).not.toThrow();
  });

  it("renders snapshot", () => {
    const tree = Renderer
    .create(<Console id="test" className="Test" key="test" logBook={[]} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });  
});
