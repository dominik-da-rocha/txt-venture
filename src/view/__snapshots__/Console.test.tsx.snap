// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console renders snapshot 1`] = `
<div
  className="Component Console Test"
  id="test"
/>
`;
